packages/eslint-plugin/lib/rules/no-unnecessary-type-assertion.js

```diff
@@ -162,6 +110,68 @@ module.exports = {
   },
 
   create(context) {
+    const sourceCode = context.getSourceCode();
+
+    /**
+     * @param {Node} node node being linted
+     * @param {Context} context linting context
+     * @param {ts.TypeChecker} checker TypeScript typechecker
+     * @returns {void}
+     */
+    function verifyCast(node, context, checker) {
```

Comment:
remove the `context` parameter.
you can just inherit it from the parent closure.

Reply:
@bradzacher that was really stupid of me, I forgot I had put that there.
In general, do we have a stylistic preference for putting functions inside create and pulling context from the closure vs. putting them outside create and passing context through?

Reply:
Hey man autopilot copy pasta is the best pasta :)

I generally lean toward putting all functions inside create. IMO if it accesses options or context it's better to put it in create so you don't have to pass them around via args.

Reply:
Sounds good. Moved everything into `create`.